sim/core/spell_outcome.go

```diff
@@ -631,27 +640,27 @@ func (spell *Spell) OutcomeExpectedMagicHitAndCrit(_ *Simulation, result *SpellR
 
 func (spell *Spell) OutcomeExpectedMeleeWhite(_ *Simulation, result *SpellResult, attackTable *AttackTable) {
 	missChance := spell.GetPhysicalMissChance(attackTable)
-	dodgeChance := TernaryFloat64(spell.Flags.Matches(SpellFlagCannotBeDodged), 0, max(0, attackTable.BaseDodgeChance - spell.ExpertisePercentage() - spell.Unit.PseudoStats.DodgeReduction))
-	parryChance := TernaryFloat64(spell.Unit.PseudoStats.InFrontOfTarget, max(0, attackTable.BaseParryChance - spell.ExpertisePercentage()), 0)
+	dodgeChance := TernaryFloat64(spell.Flags.Matches(SpellFlagCannotBeDodged), 0, max(0, attackTable.BaseDodgeChance-spell.ExpertisePercentage()-spell.Unit.PseudoStats.DodgeReduction))
+	parryChance := TernaryFloat64(spell.Unit.PseudoStats.InFrontOfTarget, max(0, attackTable.BaseParryChance-spell.ExpertisePercentage()), 0)
 	glanceChance := attackTable.BaseGlanceChance
 	blockChance := TernaryFloat64(spell.Unit.PseudoStats.InFrontOfTarget, attackTable.BaseBlockChance, 0)
 	whiteCritCap := 1.0 - missChance - dodgeChance - parryChance - glanceChance - blockChance
 	critChance := min(spell.PhysicalCritChance(attackTable), whiteCritCap)
-	averageMultiplier := 1.0 - missChance - dodgeChance - parryChance + (spell.CritMultiplier - 1) * critChance - glanceChance * (1.0 - attackTable.GlanceMultiplier)
+	averageMultiplier := 1.0 - missChance - dodgeChance - parryChance + (spell.CritMultiplier-1)*critChance - glanceChance*(1.0-attackTable.GlanceMultiplier)
 	result.Damage *= averageMultiplier
-	result.Damage -= blockChance * result.Target.BlockValue()
+	result.Damage -= blockChance * result.Target.BlockDamageReduction()
```

Comment:
I think this should be `result.Damage -= blockChance * result.Damage * result.Target.BlockDamageReduction()` if I'm understanding your change correctly